extensions/sql-database-projects/src/models/project.ts

```diff
@@ -237,11 +237,16 @@ export class Project implements ISqlProject {
 
 	private async readFolders(): Promise<FileProjectEntry[]> {
 		const folderEntries: FileProjectEntry[] = [];
+
 		// glob style getting folders for sdk style projects
+		const foldersSet = new Set<string>();
 		if (this._isSdkStyleProject) {
-			const folders = await utils.getFoldersInFolder(this.projectFolderPath, true);
-			folders.forEach(f => {
-				folderEntries.push(this.createFileProjectEntry(utils.trimUri(Uri.file(this.projectFilePath), Uri.file(f)), EntryType.Folder));
+			this.files.forEach(file => {
+				// if file is in the project's folder, add its folder. Empty folders won't be shown unless specified in the sqlproj (same as how it's handled for csproj in VS)
```

Comment:
// if file is in the project's folder, add its *a?* folder?

Reply:
updated comment to be more clear